superset/models/reports.py

```diff
@@ -0,0 +1,172 @@
+    ),
+    UniqueConstraint("user_id", "report_schedule_id"),
+)
+
+
+class ReportSchedule(Model, AuditMixinNullable):
+
+    """
+    Report Schedules, supports alerts and reports
+    """
+
+    __tablename__ = "report_schedule"
+    id = Column(Integer, primary_key=True)
+    type = Column(String(50), nullable=False)
+    label = Column(String(150), nullable=False, unique=True)
+    active = Column(Boolean, default=True, index=True)
+    crontab = Column(String(50), nullable=False)
+    sql = Column(Text())
+    # (Reports) M-O to chart
+    chart_id = Column(Integer, ForeignKey("slice.id"), nullable=True)
+    chart = relationship(Slice, backref="report_schedules", foreign_keys=[chart_id])
+    # (Reports) M-O to dashboard
+    dashboard_id = Column(Integer, ForeignKey("dashboards.id"), nullable=True)
+    dashboard = relationship(
+        Dashboard, backref="report_schedules", foreign_keys=[dashboard_id]
+    )
+    # (Alerts) M-O to database
+    database_id = Column(Integer, ForeignKey("dbs.id"), nullable=True)
+    database = relationship(Database, foreign_keys=[dashboard_id])
+    owners = relationship(security_manager.user_model, secondary=report_schedule_user)
+
+    # (Reports) email format
+    email_format = Column(String(50))
+
+    # (Alerts) Stamped last observations
+    last_eval_dttm = Column(DateTime)
+    last_state = Column(String(50))
+    last_value = Column(Float)
+    last_value_row_json = Column(Text)
+
+    # (Alerts) Observed value validation related columns
+    validator_type = Column(String(100))
+    validator_config_json = Column(Text, default="{}")
+
+    # Log retention
+    log_retention = Column(Integer, default=90)
+    grace_period = Column(Integer, default=60 * 60 * 4)
+
+    def __repr__(self) -> str:
+        return str(self.label)
+
+
+class ReportRecipients(
+    Model, AuditMixinNullable
+):  # pylint: disable=too-few-public-methods
+
+    """
+    Report Recipients, meant to support multiple notification types, eg: Slack, email
+    """
+
+    __tablename__ = "report_recipient"
+    id = Column(Integer, primary_key=True)
+    type = Column(String(50), nullable=False)
+    recipient_config_json = Column(Text, default="{}")
+    report_schedule_id = Column(
+        Integer, ForeignKey("report_schedule.id"), nullable=False
+    )
+    report_schedule = relationship(
+        ReportSchedule, backref="recipients", foreign_keys=[report_schedule_id]
+    )
+
+
+class ReportExecutionLog(Model):  # pylint: disable=too-few-public-methods
+
+    """
+    Report Execution Log, hold the result of the report execution with timestamps,
+    last observation and possible error messages
+    """
+
+    __tablename__ = "report_execution_log"
+    id = Column(Integer, primary_key=True)
+
+    # Timestamps
+    scheduled_dttm = Column(DateTime, nullable=False)
+    start_dttm = Column(DateTime)
+    end_dttm = Column(DateTime)
+
+    # (Alerts) Observed values
+    observation_dttm = Column(DateTime)
```

Comment:
isn't this effectively same as `end_dttm`?

Reply:
Yes, I think your right, let me remove that